File: com-mini-core/src/main/java/com/mini/digest/SHA512Util.java
package com.mini.digest;

import java.io.InputStream;

/**
 * SHA512Util.java
 * @author xchao
 */
public class SHA512Util {
    private static final DigestUtil instance = new DigestUtil("SHA-512");

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(String data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(String data) {
        return instance.encode(data);
    }

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(byte[] data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(byte[] data) {
        return instance.encode(data);
    }

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(InputStream data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA512加密
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(InputStream data) {
        return instance.encode(data);
    }

}

File: sn.mini.kotlin.util/src/main/java/sn/mini/kotlin/util/degest/SHA512Util.kt
/**
 * Created the com.cfinal.util.digest.CFSHA512.java
 *
 * @created 2017年5月16日 下午4:31:08
 * @version 1.0.0
 */
package sn.mini.kotlin.util.degest

import java.io.InputStream

/**
 * com.cfinal.util.digest.CFSHA512.java
 * @author XChao
 */
object SHA512Util {
    private val instance = DigestUtil("SHA-512")

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encodeToByte(data: String): ByteArray = instance.encodeToByte(data)

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encode(data: String): String = instance.encode(data)

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encodeToByte(data: ByteArray): ByteArray = instance.encodeToByte(data)

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encode(data: ByteArray): String = instance.encode(data)

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encodeToByte(data: InputStream): ByteArray = instance.encodeToByte(data)

    /**
     * SHA512加密
     * @param data
     * @return
     */
    fun encode(data: InputStream): String = instance.encode(data)

}
